include/experimental/__p1673_bits/blas3_matrix_product.hpp

```diff
@@ -558,28 +559,39 @@ template<class ElementType_A,
 void triangular_matrix_left_product(
   std::experimental::mdspan<ElementType_A, std::experimental::extents<numRows_A, numCols_A>, Layout_A, Accessor_A> A,
   Triangle t,
+  DiagonalStorage d,
   std::experimental::mdspan<ElementType_B, std::experimental::extents<numRows_B, numCols_B>, Layout_B, Accessor_B> B,
   std::experimental::mdspan<ElementType_C, std::experimental::extents<numRows_C, numCols_C>, Layout_C, Accessor_C> C)
 {
   using size_type = typename extents<>::size_type;
+  constexpr bool explicitDiagonal =
+    std::is_same_v<DiagonalStorage, explicit_diagonal_t>;
 
   if constexpr (std::is_same_v<Triangle, lower_triangle_t>) {
     for (size_type j = 0; j < C.extent(1); ++j) {
       for (size_type i = 0; i < C.extent(0); ++i) {
         C(i,j) = ElementType_C{};
-        for (size_type k = 0; k <= i; ++k) {
+        const size_type k_upper = explicitDiagonal ? i : i - size_type(1);
```

Comment:
`size_type` is unsigned, so  if `i` is zero and `explicitDiagonal` is `false`, then `i - size_type(i)` will be a big positive number.  I think the fix would be to change the bounds of the for loop over `i`.

Reply:
Done!